### PR TITLE
Adopt Contributor Covenant Code of Conduct version 2.1

### LINKED ISSUE #403: Expand procedure for enforcing code of conduct
As our community grows, it will be important to expand [our code of conduct enforcement procedure](https://github.com/PlasmaPy/PlasmaPy/blob/40005195fd0b763b040a6252dc1fb8d92bd9197e/CODE_OF_CONDUCT.md#enforcement), especially before we host any conferences.  Some possibilities include:
 - [ ] Create and link to an email address (preferably from the `@plasmapy.org` domain) that code of conduct violations can be reported to.
 - [ ] State that complaints can be submitted to any Coordinating Committee member.
 - [ ] Describe the process by which Coordinating Committee members investigate and respond to complaints that occurred during the course of remote code development.
 - [ ] Describe the process by which complaints are investigated and responded to during in-person development meetings and conferences.
 - [ ] State that Coordinating Committee members with significant conflicts of interest should recuse themselves.
 - [ ] Appoint an ombudsperson?  (a long-term possibility that follows Astropy's example)

This issue will need a lot of input from the Coordinating Committee and the broader PlasmaPy community, so it will need to be brought up during our biweekly meetings as we are working on it.

An example procedure is the sci-fi convention [Readercon's code of conduct enforcement procedure](http://readercon.org/safety/procedures.htm).  This reason why I remember this one is that [Readercon messed up horrendously](http://readercon.org/publicstatement.htm) in enforcing it a few years ago, after which they redid their policy completely.

### LINKED ISSUE #853: Incorporate Contributor Covenent v2.0 into our code of conduct?
[Version 2.0 of the Contributor Covenant](https://www.contributor-covenant.org/version/2/0/code_of_conduct/) code of conduct was recently released. It looks like a major improvement upon [version 1.4](https://www.contributor-covenant.org/version/1/4/code-of-conduct/). We incorporated part of version 1.4 as well as [Astropy's community code of conduct](https://www.astropy.org/code_of_conduct.html) in [our code of conduct](https://docs.plasmapy.org/en/latest/CODE_OF_CONDUCT.html).  The questions for us are:

- Are there parts of the Contributor Covenant v2.0 that we should incorporate into our code of conduct?
- Should we adopt the Contributor Covenant v2.0 as our code of conduct since it has become an open source community standard?

If we address this, we could also address #403.

### PR DESCRIPTION
**Pull request by namurphy**
This PR is to update our code of conduct to the [Contributor Covenant Code of Conduct v2.1](https://www.contributor-covenant.org/version/2/1/code_of_conduct/).  

Our current code of conduct is a hybrid between Astropy's and an older version of the Contributor Covenant.  One of the things missing from our current code of conduct is an enforcement policy (#403).  Versions 2.0 and 2.1 of the Contributor Covenant do an enforcement policy, which is a significant improvement on our current code of conduct.  The Contributor Covenant is also effectively becoming the standard code of conduct for open source projects, and it was even the code of conduct that we adopted for Plasma Hack Week.  The Contributor Covenant is also time-tested and community-maintained.   This PR would need the approval of the Coordinating Committee, I think.

Remaining tasks...

- [x] Create the contact email address for the code of conduct
- [x] Check that the documentation builds correctly
- [x] Add a badge to `README.md` (following instructions on the Contributor Covenant website)
- [x] Add a changelog entry

Closes #403.  Closes #853.  This follows up on #854.

### CODE REVIEW

**Review by StanczakDominik — APPROVED** [review]
That did not diff well at all. 😅 

Looks good to me, though!

File: docs/CODE_OF_CONDUCT.rst [hunk 1]
```diff
@@ -1,127 +1,145 @@
...
-Project team members should recuse themselves from enforcement of the
-code of conduct for a given incident if they have an actual or apparent
-conflict of interest.  Further details of specific enforcement policies
-may be posted separately.
-
-Project maintainers who do not follow or enforce the Code of Conduct in
-good faith may face temporary or permanent repercussions as determined
-by other members of the project’s leadership.
+be reported to the community leaders responsible for enforcement at
+``conduct@plasmapy.org``. All complaints will be reviewed and
```
Comment (rocco8773) [inline docs/CODE_OF_CONDUCT.rst:72]: Does `sphinx` have an email role?

Comment (rocco8773) [reply docs/CODE_OF_CONDUCT.rst:72]: Possible solution...

https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#external-links

Comment (rocco8773) [reply docs/CODE_OF_CONDUCT.rst]: Another possible solution...

https://github.com/sphinx-contrib/email

File: docs/CODE_OF_CONDUCT.rst [hunk 2]
```diff
@@ -1,127 +1,145 @@
...
-=========================
-
-The PlasmaPy community strives to follow the best practices in open
-source software development. New contributors are encouraged to join the
-team and contribute to the codebase. We anticipate/encourage a global
-participation from people with diverse backgrounds, skills, interests,
-and opinions. We believe that such diversity is critical in ensuring a
-growth of ideas in our community.
+Contributor Covenant Code of Conduct
+====================================
```
Comment (rocco8773) [inline docs/CODE_OF_CONDUCT.rst:4]:
```suggestion
Contributor Covenant Code of Conduct
====================================

We adopt `Contributor Covenant 2.1 <https://www.contributor-covenant.org/version/2/1/code_of_conduct/>`_ which is reproduced below.
```

File: changelog/1324.doc.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Adopted the `Contributor Covenant Code of Conduct version 2.1
+<https://www.contributor-covenant.org/version/2/1/code_of_conduct/>`__.
```
Comment (rocco8773) [inline changelog/1324.doc.rst:2]:
```suggestion
Adopted the `Contributor Covenant Code of Conduct version 2.1
<https://www.contributor-covenant.org/version/2/1/code_of_conduct/>`__
and updated the
:ref:`Contributor Covenant Code of Conduct <plasmapy-code-of-conduct>`
page accordingly.
```

File: docs/CODE_OF_CONDUCT.rst [hunk 4]
```diff
@@ -1,127 +1,148 @@
...
-Project team members should recuse themselves from enforcement of the
-code of conduct for a given incident if they have an actual or apparent
-conflict of interest.  Further details of specific enforcement policies
-may be posted separately.
-
-Project maintainers who do not follow or enforce the Code of Conduct in
-good faith may face temporary or permanent repercussions as determined
-by other members of the project’s leadership.
+be reported to the community leaders responsible for enforcement at
+conduct@plasmapy.org. All complaints will be reviewed and
```
Comment (namurphy) [inline docs/CODE_OF_CONDUCT.rst:75]: It turns out that Sphinx will render email addresses correctly with a `mailto:` prefix if they show up in plain text.

**Comment by namurphy** [comment]
Earlier this afternoon I incorporated @rocco8773's suggestions and made a few other slight changes, so this is ready for review again.  I also created the conduct at plasmapy.org email group.  Thanks!

**Review by StanczakDominik — DISMISSED** [review]

**Review by rocco8773 — APPROVED** [review]

**Pull request merged**